Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -420,6 +428,10 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
     if (m_compressedSampleQueue.isEmpty())
         return;
 
+    if (decodedSamplesCount() > maximumDecodedSampleCount(*decompressionSession))
+        return;
+
+    m_compressedSampleQueueSize--;
```

Comment:
It seems possible that `m_compressedSampleQueueSize` is 0 here. Say if we call `flushCompressedSampleQueue()` in the main thread just before calling `decodeNextSample()` in the dispatcher queue? We would end up with a very big value, not sure whether that has practical impact but it seems best to avoid this.

I would tend to simplify a bit `m_compressedSampleQueueSize` by doing one of these options:
- incrementing/decrementing this value only in dispatcher queue, when appending/taking a sample from `m_compressedSampleQueue`
- Setting `m_compressedSampleQueueSize` to `m_compressedSampleQueue.size()` here if it is important to set aggressively `m_compressedSampleQueueSize` in main thread.

Reply:
If we do not increase m_compressedSampleQueueSize on the main thread, then `isReadyForMoreMediaData()` will return true until the lambda actually runs on the dispatcher queue ; this can lead to much more data being fed to the VideoMediaSampleRenderer than we want. 

Option 2 seems the most reasonable.